facades/WorkspaceValidation/src/test/java/org/terasology/workspace/validation/WorkspaceModulesResolvingTests.java

```diff
@@ -0,0 +1,56 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.workspace.validation;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.terasology.engine.core.module.ModuleManager;
+import org.terasology.gestalt.naming.Name;
+
+import java.util.Set;
+import java.util.stream.Stream;
+
+/**
+ * Example test for  check modules dependency validations.
+ */
+public class WorkspaceModulesResolvingTests {
+
+    public static Stream<Arguments> modulesAndModuleManager() {
+        System.setProperty(ModuleManager.LOAD_CLASSPATH_MODULES_PROPERTY, "true");
+        ModuleManager temporary = new ModuleManager("");
+        return temporary.getRegistry()
+                .getModuleIds()
+                .stream()
+                .map((name) -> Arguments.of(temporary, name));
+    }
+
+    public static Stream<Arguments> modulePairsAndModuleManager() {
+        System.setProperty(ModuleManager.LOAD_CLASSPATH_MODULES_PROPERTY, "true");
+        ModuleManager temporary = new ModuleManager("");
+        Set<Name> moduleIds = temporary.getRegistry()
+                .getModuleIds();
```

Comment:
Hmm, I have been trying to deprecate ModuleManager#getRegistry, because for the most part, other things being able to mess with the registry that ModuleManager is responsible for managing is bad for business.

Here you are using it to get all the modules in the current workspace. We could add a method to return this sort of read-only collection of module IDs, that would be pretty safe.

I'm guessing the other use case we have for wanting to see all the workspace's modules is the Advanced Game Setup screen where you pick which modules to activate. That uses a lot more than just the ID... I haven't looked at it to find out what interface it really wants.